dist/module/cardDealer.mjs

```diff
@@ -53,15 +53,15 @@ class CardDealer {
             const pile = this.pile;
 
             // Deal 1 random card and grab reference to the dealt card
-            await deck.deal([pile], 1, { how: CONST.CARD_DRAW_MODES.RANDOM, action: shareToAll ? 'deal orcnog_card_viewer_doshare' : 'deal' });
+            await deck.deal([pile], 1, { how: CONST.CARD_DRAW_MODES.RANDOM });
```

Comment:
Hey @kristianserrano , thank you for this PR.

Unfortunately, re: this particular line, modifying the "action" attribute was the only mechanism I found to tag a `.deal` event so I can key off of it an know whether it's appropriate to share to all, or to whisper details to the DM's chat.  There are scenarios where I don't want to do one or both of these.

Can you explain what issue this is causing in v10?  Is it totally crashing the `.deal` call? Or is it just throwing a console error but still working?

Reply:
I figured that was the case. I'll think about other ways we can pass that tag. Perhaps a mutable property on the deal instance?

And yes it was crashing the whole function because it hits against a `chatActions` object in Foundry VTT that has fixed properties (pass, deal, etc), so it looks for the key via `chatActions[action]` and doesn't match because `action` is your custom string.

Reply:
@orcnog regarding the action context, one possible solution might be to set and get a hidden world setting with a `config` set to `false` during the execution. The value can be an object of properties with true/false values for determining behaviors. Another option might be a global variable.

Reply:
@orcnog I've added two new hidden world settings for share and view. I replaced the use of the custom action strings with setting and getting those settings.